core/modules/rproc/InfileMerger.h

```diff
@@ -107,7 +107,7 @@ class InfileMergerConfig {
 class InvalidJobAttemptMgr {
 public:
     InvalidJobAttemptMgr() {}
-    void setDeleteFunc(std::function<bool(int)> func) {_deleteFunc = func; }
+    void setDeleteFunc(std::function<bool(std::set<int> const& jobAttempts)> func) {_deleteFunc = func; }
```

Comment:
Since this definition (of a function) appears on code at least 3 times (the first time is here, the second time is in a definition of the data member, and the third time is in the .CC file) then I recommend introducing a typedef:

  typedef std::function<bool(std::set\<int\> const&)> deleteFuncType;

This will make the code a bit easier to read.